master/internal/stream/system.go

```diff
@@ -295,7 +369,12 @@ func doPublishLoop[T stream.Msg](
 		// The pq listener example includes a timeout case, so we do too.
 		// (https://pkg.go.dev/github.com/lib/pq/example/listen)
 		case <-time.After(30 * time.Second):
-			go listener.Ping()
+			go func() {
+				err = listener.Ping()
+				if err != nil {
+					log.Error(errors.Wrap(err, "no active connection"))
```

Comment:
hm, it seems like just logging the error can't be the right answer

Reply:
would you suggest returning the error back up to the `publishLoop` and let it attempt a restart?